Local build testing broken on Fedora 35 by #21154.

Testing fails locally when checking `long double` with
```
ERROR numpy/core/tests/test_longdouble.py - UserWarning: Signature b'\x00\xd0\xcc\xcc\xcc\xcc\xcc\xcc\xfb\xbf\xff\x7f\x00\x00\x00\x00...
ERROR numpy/core/tests/test_multiarray.py - KeyboardInterrupt
ERROR numpy/core/tests/test_scalar_methods.py - UserWarning: Signature b'\x00\xd0\xcc\xcc\xcc\xcc\xcc\xcc\xfb\xbf\xff\x7f\x00\x00\x00...
ERROR numpy/core/tests/test_scalarmath.py - UserWarning: Signature b'\x00\xd0\xcc\xcc\xcc\xcc\xcc\xcc\xfb\xbf\xff\x7f\x00\x00\x00\x00...
ERROR numpy/core/tests/test_umath.py - UserWarning: Signature b'\x00\xd0\xcc\xcc\xcc\xcc\xcc\xcc\xfb\xbf\xff\x7f\x00\x00\x00\x00' for...

```

There are three problems I can see:

1. The warning was not issued previous to #21154.
2. The warning should be ignored, maybe should not be issued (`numpy/core/getlimits.py` line 346).
3. Why is CI not failing? 

EDIT: Ignoring the warning leads to actual errrors:
```
ERROR numpy/core/tests/test_longdouble.py - AttributeError: 'MachAr' object has no attribute '_str_smallest_normal'
ERROR numpy/core/tests/test_scalar_methods.py - AttributeError: 'MachAr' object has no attribute '_str_smallest_normal'
ERROR numpy/core/tests/test_scalarmath.py - AttributeError: 'MachAr' object has no attribute '_str_smallest_normal'
ERROR numpy/core/tests/test_umath.py - AttributeError: 'MachAr' object has no attribute '_str_smallest_normal'
```

EDIT2: gcc (GCC) 11.3.1 20220421 (Red Hat 11.3.1-2)

I get some errors in `numpy/typing/tests/test_typing.py::test_reveal` but the test suite runs:

<details>
<summary>Details</summary>



```
$ git checkout 3f9fada45852a8f1b3fd9e0b760f34425099896c
<Blah blah blah detached head>
$ python runtests.py -m'full'
Building, see build.log...
    ... build in progress
Build OK
/home/hood/Documents/programming/numpy/build/testenv/lib/python3.9/site-packages/numpy/_pytesttester.py:145: DeprecationWarning: 

  `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
  of the deprecation of `distutils` itself. It will be removed for
  Python >= 3.12. For older Python versions it will remain present.
  It is recommended to use `setuptools < 60.0` for those Python versions.
  For more details, see:
    https://numpy.org/devdocs/reference/distutils_status_migration.html 

<... test progress info ...>

  from numpy.distutils import cpuinfo
NumPy version 1.23.0.dev0+974.g3f9fada45
NumPy relaxed strides checking option: True
NumPy CPU features:  SSE SSE2 SSE3 SSSE3* SSE41* POPCNT* SSE42* AVX* F16C* FMA3* AVX2* AVX512F* AVX512CD* AVX512_KNL? AVX512_KNM? AVX512_SKX* AVX512_CLX* AVX512_CNL* AVX512_ICL*

...

FAILED numpy/typing/tests/test_typing.py::test_reveal[numeric] - AssertionError: Reveal mismatch at line 108
FAILED numpy/typing/tests/test_typing.py::test_reveal[arrayterator] - AssertionError: Reveal mismatch at line 12
FAILED numpy/typing/tests/test_typing.py::test_reveal[stride_tricks] - AssertionError: Reveal mismatch at line 24
FAILED numpy/typing/tests/test_typing.py::test_reveal[ndarray_misc] - AssertionError: Reveal mismatch at line 168
FAILED numpy/typing/tests/test_typing.py::test_reveal[dtype] - AssertionError: Reveal mismatch at line 51
FAILED numpy/typing/tests/test_typing.py::test_reveal[nditer] - AssertionError: Reveal mismatch at line 10
FAILED numpy/typing/tests/test_typing.py::test_reveal[array_constructors] - AssertionError: Reveal mismatch at line 171
FAILED numpy/typing/tests/test_typing.py::test_reveal[flatiter] - AssertionError: Reveal mismatch at line 8
FAILED numpy/typing/tests/test_typing.py::test_reveal[multiarray] - AssertionError: Reveal mismatch at line 37
FAILED numpy/typing/tests/test_typing.py::test_reveal[fromnumeric] - AssertionError: Reveal mismatch at line 122
FAILED numpy/typing/tests/test_typing.py::test_reveal[testing] - AssertionError: Reveal mismatch at line 33
FAILED numpy/typing/tests/test_typing.py::test_reveal[index_tricks] - AssertionError: Reveal mismatch at line 19
12 failed, 20000 passed, 168 skipped, 31 xfailed, 3 xpassed in 431.85s (0:07:11)
```



</details>

The problem is probably compiler related:
```
INFO: gcc: _configtest.c
_configtest.c: In function ‘main’:
_configtest.c:9:3: error: argument 1 null where non-null expected [-Werror=nonnull]
    9 |   strtold_l((const char*) 0, (char**) 0, (locale_t) 0);
      |   ^~~~~~~~~
In file included from _configtest.c:1:
/usr/include/stdlib.h:303:20: note: in a call to function ‘strtold_l’ declared ‘nonnull’
  303 | extern long double strtold_l (const char *__restrict __nptr,
      |                    ^~~~~~~~~
_configtest.c:9:3: error: argument 3 null where non-null expected [-Werror=nonnull]
    9 |   strtold_l((const char*) 0, (char**) 0, (locale_t) 0);
      |   ^~~~~~~~~
In file included from _configtest.c:1:
/usr/include/stdlib.h:303:20: note: in a call to function ‘strtold_l’ declared ‘nonnull’
  303 | extern long double strtold_l (const char *__restrict __nptr,
      |                    ^~~~~~~~~
```

--- a/numpy/core/feature_detection_locale.h
+++ b/numpy/core/feature_detection_locale.h
@@ -1,1 +1,2 @@
+#pragma GCC diagnostic ignored "-Wnonnull"
 long double strtold_l(const char*, char**, locale_t);

